### ISSUE TITLE
Automatically purge packages one on the install list if the package is no longer in use

### ISSUE THREAD

**Issue by JohRest** [mylinuxforwork/dotfiles#200]
I'd like to propose a new concept of handling no longer needed packages that in the past were installed.

Example: ml4w uses (at the moment) starship for prompt generation, but decides (in the future) to use oh-my-posh instead (due to good reasons)

Then oh-my-posh is built and installed, but (in this case) starship remains as a (no longer) used package on the system.

So if ml4w switches to a new package wouldn't it be good to purge it (with existence check and only removes the package via package manager [yay etc.]) from the user system?

**Comment by mylinuxforwork** [comment]
I understand your comment but that's difficult to handle in general.

Maybe the ML4W Dotfiles are installed in parallel to another DE and some packages are still needed. 

Or two parallel Dotfiles configurations are installed into different folders.

Not sure if your request is easily possible to solve.

**Comment by JohRest** [comment]
This was more of a discussion attempt than a real enhancement. I am fully aware the over-caring packages related to ml4w might cause problems to the underlying OS if it purges packages because it thinks it is no longer needed. It might still be needed by the user. So I will close this topic and do the necessary cleanup myself. It would be godd though to have some sort of announcement when we switch packages (i.e. going from starship to oh-my-posh), so that the user is aware of the change and can decide whether to remove the starship bits from his/her system...

**Comment by mylinuxforwork** [comment]
> This was more of a discussion attempt than a real enhancement. I am fully aware that over-caring packages related to ml4w might cause problems to the underlying OS if it purges packages because ml4w thinks they are no longer needed. It might still be needed by the user. So I will close this topic and do the necessary cleanup myself. It would be godd though to have some sort of announcement when we switch packages (i.e. going from starship to oh-my-posh), so that the user is aware of the change and can decide whether to remove the starship bits from his/her system...

Thanks for your feedback.
